### PR TITLE
Support submitting a Client form with MCI value specified

### PR DESCRIPTION
**Pull request by gigxz**

### CODE REVIEW

File: drivers/hmis/app/models/hmis/hud/processors/client_processor.rb [hunk 1]
```diff
@@ -30,6 +36,9 @@ def process(field, value)
         # If hidden due to permissions, set to old value or 99
         attribute_value = @processor.send(factory_name).dob_data_quality || 99 if value == Base::HIDDEN_FIELD_VALUE
         { attribute_name => attribute_value }
+      when 'mci_id'
+        process_mci(value)
+        {}
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/processors/client_processor.rb:41]: I don't love this as a pattern, but it's quick. Maybe when we implement the "custom field" attribute and processor, we should move this logic there. (Even though it won't be stored as a custom field).

Comment (eanders) [reply drivers/hmis/app/models/hmis/hud/processors/client_processor.rb:41]: What's this doing?  It looks like there are various options, one of which will find a client and then mark `client.create_mci_id` to true but do nothing more?  Does that then get processed further down the stack?

**Review by eanders — APPROVED** [review]

File: drivers/hmis/app/models/hmis/hud/processors/client_processor.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/models/hmis/hud/processors/client_processor.rb:41]: Yeah. So `process_mci` will process a form value that is one of `_CREATE_MCI_ID`, `_CREATE_UNCLEARED_CLIENT` or `<mci id to link>`.

The special logic for creating a new MCI ID gets processed in the `after_save` hook on Client. It's the same pattern we use for image uploads. I find it kind of hard to follow, since it's split across multiple files, but I couldn't come up with a better pattern to achieve the desired behavior:

1. We should only create the MCI ID if the HMIS client was successfully validated/saved.
2. We should only create the HMIS Client if the new MCI ID was actually successfully created. (`after_save` is called in the same transaction as `save` so we get this).

The code in this PR achieves that^ but maybe it's not the clearest pattern.. is there a more straightforward way to express that?
